src/hooks/useTodoItemCardComplete.ts

```diff
@@ -1,19 +1,19 @@
 import axios, { AxiosError } from "axios";
 import { useState } from "react";
-import { ICard } from "../components/Card";
+import { ITodoItemCard } from "../components/todo-item-card";
 
-const useCardComplete = () => {
+const useTodoItemCardComplete = () => {
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState("");
 
-  async function cardComplete(card: ICard) {
+  async function todoItemCardComplete(todoItemCard: ITodoItemCard) {
     try {
       setError("");
       setIsLoading(true);
-      card.isCompleted = true;
-      await axios.put<ICard>(
-        "https://aufgabenliste.azurewebsites.net/api/todo" + "/" + card.id + "?id=" + card.id,
-        card)
+      todoItemCard.isCompleted = true;
```

Comment:
Why you are changing the incoming parameter? It has to be immutable.  Dirty hack.

Reply:
I don't know how to re-write this field after getting clicked status on the button... or ... should I take the whole model from the form ?